api/src/main/java/io/serverlessworkflow/api/deserializers/DataInputSchemaDeserializer.java

```diff
@@ -46,17 +53,44 @@ public DataInputSchema deserialize(JsonParser jp, DeserializationContext ctxt)
     JsonNode node = jp.getCodec().readTree(jp);
 
     DataInputSchema dataInputSchema = new DataInputSchema();
+    JsonNode schemaDefinition = null;
 
-    if (!node.isObject()) {
-      dataInputSchema.setSchema(node.asText());
-      dataInputSchema.setFailOnValidationErrors(true); // default
-
-      return dataInputSchema;
-    } else {
-      dataInputSchema.setSchema(node.get("schema").asText());
+    if (node.isObject() && node.get("schema").isObject() && !node.get("schema").isEmpty()) {
+      schemaDefinition = node.get("schema");
       dataInputSchema.setFailOnValidationErrors(node.get("failOnValidationErrors").asBoolean());
+    } else {
+      String schemaFileDef = node.isObject() ? node.get("schema").asText() : node.asText();
+      dataInputSchema.setFailOnValidationErrors(true);
+      dataInputSchema.setRefValue(schemaFileDef);
+      String schemaFileContent = Utils.getResourceFileAsString(schemaFileDef);
```

Comment:
Although not related with this PR, the attempt to load the content of the schema from the URI is only partially implemented (because it does not support file or http, just classpath) and I think can be skipped. 
We can just say that is responsability of the implementation to load the schema from the uri if the refValue is not null,
we simpligy the whole thing, either we have refvalue set to an uri, either we have the content read from the inlined schema, what we have now is really confusing.

Reply:
I agree since we are doing the same for the OpenAPI, Async, and gRPC functions (not loading the resources). We can have an implementation later that would fetch from classpath or other sources and have the possibility to let the runtime to override this behavior if needed.